script.js

```diff
@@ -1,4 +1,456 @@
+        gameMode = "KNOCKOUT";
+      } else {
+        gameMode = "RANK";
+      }
+
+      myOutputValue = `You have selected the ${input} mode.<br><br>
+      Please select the version that you would like to play by entering
+        "Normal" or "Lowest Combined"`;
+
+      return myOutputValue;
+    } else if (gameStatus == "version selection") {
+      gameStatus = "player selection";
+
+      if (input.toUpperCase() == "LOWEST COMBINED") {
+        gameVersion = "lowestCombined";
+      } else {
+        gameVersion = "normal";
+      }
+
+      myOutputValue = `You have selected the ${input} version.<br><br>
+      Please enter the number of players.`;
+
+      return myOutputValue;
+    } else if (gameStatus == "player selection") {
+      numOfPlayers = input;
+      createPlayersArray(input);
+      gameStatus = "dice selection";
+
+      myOutputValue = `You have selected ${input} players.<br><br>
+      Please enter the number of dice to play.`;
+
+      return myOutputValue;
+    } else if (gameStatus == "dice selection") {
+      numOfDice = input;
+      gameStatus = "game intro";
+
+      myOutputValue = `You have selected ${input} dice.<br><br>
+      Please click submit to start rolling!`;
+
+      return myOutputValue;
+    } else if (gameStatus == "game intro") {
+      if (gameMode == "RANK") {
+        return playGame(numOfPlayers);
+      } else if (gameMode == "KNOCKOUT") {
+        return playKnockoutGame(numOfPlayers);
+      }
+    }
+  } else {
+    return inputValidationResults;
+  }
+};
+
+// Function for input validation
+
+var inputValidation = function (input) {
+  var returnStatement = "";
+  if (gameStatus == "version selection") {
+    if (
+      !(
+        input.toUpperCase() == "NORMAL" ||
+        input.toUpperCase() == "LOWEST COMBINED"
+      )
+    ) {
+      returnStatement = `Wrong input. Please enter only "Normal" or "Lowest Combined" to start the game.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  } else if (gameStatus == "game intro" || gameStatus == "player 2 roll dice") {
+    return true;
+  } else if (gameStatus == "player selection") {
+    if (Number.isNaN(Number(input)) == true || input < 2 || input == "") {
+      returnStatement = `Wrong input. Please enter a number greater or equal to 2.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  } else if (gameStatus == "dice selection") {
+    if (Number.isNaN(Number(input)) == true || input == 0 || input == "") {
+      returnStatement = `Wrong input. Please enter a number greater than 0.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  } else if (gameStatus == "mode selection") {
+    if (
+      !(input.toUpperCase() == "KNOCKOUT" || input.toUpperCase() == "ORIGINAL")
+    ) {
+      returnStatement = `Wrong input. Please enter only "Original" or "Knockout" to start the game.`;
+      return returnStatement;
+    } else {
+      return true;
+    }
+  }
```

Comment:
If you find yourself using multiple else if statements, you might want to look at using a [Switch](https://www.w3schools.com/js/js_switch.asp) statement instead of an if/else.